File: SDKs/googleapis.com/apigateway/v1beta/python/src/sdk/models/shared/apigatewaylistlocationsresponse.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import apigatewaylocation as shared_apigatewaylocation


@dataclass_json
@dataclasses.dataclass
class ApigatewayListLocationsResponse:
    r"""ApigatewayListLocationsResponse
    The response message for Locations.ListLocations.
    """
    
    locations: Optional[list[shared_apigatewaylocation.ApigatewayLocation]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('locations') }})
    next_page_token: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('nextPageToken') }})
    

File: SDKs/googleapis.com/apigateway/v1/java/lib/src/main/java/openapisdk/models/shared/ApigatewayListLocationsResponse.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * ApigatewayListLocationsResponse
 * The response message for Locations.ListLocations.
**/
public class ApigatewayListLocationsResponse {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("locations")
    public ApigatewayLocation[] locations;
    public ApigatewayListLocationsResponse withLocations(ApigatewayLocation[] locations) {
        this.locations = locations;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("nextPageToken")
    public String nextPageToken;
    public ApigatewayListLocationsResponse withNextPageToken(String nextPageToken) {
        this.nextPageToken = nextPageToken;
        return this;
    }
}